statsmodels/tsa/vector_ar/vecm.py

```diff
@@ -641,11 +641,14 @@ def resid(y, x):
 
     dx = detrend(dx, f)
     r0t = resid(dx, z)
-    lx = endog[:-k_ar_diff]
+    # GH 5731, [:-0] does not work, need [:t-0]
+    lx = np.asarray(endog)[:endog.shape[0]-k_ar_diff]
```

Comment:
there should be an `endog = np.asarray(endog)` at the beginning of the function
It's missing and pandas Series might not work correctly